File: app/src/main/java/com/example/nt118/MainActivity.java
package com.example.nt118;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.example.nt118.Deadline.DeadlineActivity;
import com.google.android.material.bottomnavigation.BottomNavigationView;

import java.util.ArrayList;
import java.util.List;

public class MainActivity extends AppCompatActivity {

    private TextView[] dayViews;
    private int selectedDayIndex = 1; // Default to day 07 (index 1)
    private RecyclerView rvClassList;
    private ClassAdapter classAdapter;
    private List<ClassModel> classList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home_course);

        BottomNavigationView bottomNavigationView = findViewById(R.id.bottom_navigation);

        bottomNavigationView.setOnItemSelectedListener(item -> {
            int itemId = item.getItemId();
            if (itemId == R.id.nav_profile) {
                Intent intent = new Intent(this, ProfileActivity.class);
                startActivity(intent);
                overridePendingTransition(0, 0);
                return true;
            }
            else if (itemId == R.id.nav_deadline) {
                Intent intent = new Intent(this, DeadlineActivity.class);
                startActivity(intent);
                overridePendingTransition(0, 0);
                return true;
            }
            return false;
        });

        // Initialize day views
        dayViews = new TextView[]{
                findViewById(R.id.day06),
                findViewById(R.id.day07),
                findViewById(R.id.day08),
                findViewById(R.id.day09),
                findViewById(R.id.day10),
                findViewById(R.id.day11),
                findViewById(R.id.day12)
        };

        // Setup RecyclerView
        rvClassList = findViewById(R.id.rvClassList);
        rvClassList.setLayoutManager(new LinearLayoutManager(this));
        classAdapter = new ClassAdapter(classList, this);
        rvClassList.setAdapter(classAdapter);

        // Set click listeners for days
        for (int i = 0; i < dayViews.length; i++) {
            final int index = i;
            dayViews[i].setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View v) {
                    selectDay(index);
                }
            });
        }

        // Initial load of day 7 classes
        updateClassInfo(selectedDayIndex);
    }

    private void selectDay(int index) {
        // Reset previous selected day
        dayViews[selectedDayIndex].setBackgroundResource(0);
        dayViews[selectedDayIndex].setTextColor(getResources().getColor(android.R.color.black));

        // Set new selected day
        selectedDayIndex = index;

        // Apply appropriate styling based on the day
        if (index == 1) { // Day 07 - Blue circle
            dayViews[selectedDayIndex].setBackgroundResource(R.drawable.circle_background_blue);
            dayViews[selectedDayIndex].setTextColor(getResources().getColor(android.R.color.white));
        } else if (index == 6) { // Day 12 - Blue border
            dayViews[selectedDayIndex].setBackgroundResource(R.drawable.circle_border_blue);
            dayViews[selectedDayIndex].setTextColor(getResources().getColor(R.color.blue));
        } else { // Other days - Normal
            dayViews[selectedDayIndex].setBackgroundResource(R.drawable.circle_background_blue);
            dayViews[selectedDayIndex].setTextColor(getResources().getColor(android.R.color.white));
        }

        // Update class list for the selected day
        updateClassInfo(index);
    }

    private void updateClassInfo(int dayIndex) {
        // Clear previous class list
        classList.clear();

        // Based on the day index, load different classes
        // This is just example data - in a real app, you would load from database or API
        if (dayIndex == 1) { // Monday (07)
            classList.add(new ClassModel("1:00", "4:00", "Introduction to Programming (Lecture)", "C201"));
        } else if (dayIndex == 2) { // Tuesday (08)
            classList.add(new ClassModel("7:30", "10:45", "Introduction to UX Design (Lecture)", "C313"));
        } else if (dayIndex == 3) { // Wednesday (09)

        } else if (dayIndex == 4) { // Thursday (08)
            classList.add(new ClassModel("7:30", "9:45", "Database (Lecture)", "B304"));
            classList.add(new ClassModel("1:00", "4:00", "Database (Practice)", "B204"));
        } else if (dayIndex == 0) { // Sunday (06)
            // No classes on Sunday
        } else if (dayIndex == 5) { // Friday (09)
            classList.add(new ClassModel("7:30", "10:45", "Statistical Probability", "C313"));
        } else { // Other days

        }

        // Notify adapter that data has changed
        classAdapter.notifyDataSetChanged();
    }
}

File: app/src/main/java/com/example/e_learningapp/MainActivity.kt
package com.example.e_learningapp

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_login)

        // Ánh xạ các thành phần
        val etEmail = findViewById<EditText>(R.id.etEmail)
        val etPassword = findViewById<EditText>(R.id.etPassword)
        val btnLogin = findViewById<Button>(R.id.btnLogin)
        val tvForgotPassword = findViewById<TextView>(R.id.tvForgotPassword)

        // Xử lý sự kiện nhấn nút đăng nhập
        btnLogin.setOnClickListener {
            val email = etEmail.text.toString().trim()
            val password = etPassword.text.toString().trim()

            // Kiểm tra dữ liệu nhập
            if (email.isEmpty() || password.isEmpty()) {
                Toast.makeText(this, "Vui lòng nhập đầy đủ thông tin", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            // Nếu cần, kiểm tra định dạng email
            if (!android.util.Patterns.EMAIL_ADDRESS.matcher(email).matches()) {
                Toast.makeText(this, "Email không hợp lệ", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }

            // Gọi hàm xử lý đăng nhập (có thể là gọi API hoặc kiểm tra cục bộ)
            performLogin(email, password)
        }

        // Xử lý sự kiện “Quên mật khẩu”
        tvForgotPassword.setOnClickListener {
            // Chuyển sang màn hình quên mật khẩu hoặc hiển thị thông báo
            Toast.makeText(this, "Chức năng quên mật khẩu chưa được triển khai", Toast.LENGTH_SHORT).show()
        }
    }

    // Hàm xử lý đăng nhập
    private fun performLogin(email: String, password: String) {
        // Giả lập xử lý đăng nhập, có thể thay bằng API thực tế
        if (email == "test@example.com" && password == "123456") {
            Toast.makeText(this, "Đăng nhập thành công!", Toast.LENGTH_SHORT).show()
            // Chuyển sang Activity chính của ứng dụng
        } else {
            Toast.makeText(this, "Đăng nhập thất bại, kiểm tra lại thông tin", Toast.LENGTH_SHORT).show()
        }
    }
}
